tests/unit/core/sagas/test_user.js

```diff
@@ -0,0 +1,63 @@
+import SagaTester from 'redux-saga-tester';
+
+import userSaga from 'core/sagas/user';
+import userReducer, { userProfileLoaded } from 'core/reducers/user';
+import * as api from 'core/api/user';
+import apiReducer from 'core/reducers/api';
+import { setAuthToken } from 'core/actions';
+import { dispatchSignInActions } from 'tests/unit/amo/helpers';
+import { createUserProfileResponse, userAuthToken } from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  let mockApi;
+  let sagaTester;
+  let rootTask;
+
+  beforeEach(() => {
+    mockApi = sinon.mock(api);
+    const initialState = dispatchSignInActions().state;
+    sagaTester = new SagaTester({
+      initialState,
+      reducers: {
+        api: apiReducer,
+        user: userReducer,
+      },
+    });
+    rootTask = sagaTester.start(userSaga);
+  });
+
+  it('calls the API to fetch user profile after setAuthToken()', async () => {
+    const profile = createUserProfileResponse();
+
+    mockApi
+      .expects('userProfile')
+      .once()
+      .returns(Promise.resolve(profile));
+
+    sagaTester.dispatch(setAuthToken(userAuthToken()));
+
+    const expectedCalledAction = userProfileLoaded({ profile });
+
+    await sagaTester.waitFor(expectedCalledAction.type);
+    mockApi.verify();
+
+    const calledAction = sagaTester.getCalledActions()[1];
+    expect(calledAction).toEqual(expectedCalledAction);
+  });
+
+  it('lets exceptions to be thrown', () => {
+    const error = new Error('some API error maybe');
+    mockApi
+      .expects('userProfile')
+      .returns(Promise.reject(error));
+
+    sagaTester.dispatch(setAuthToken(userAuthToken()));
+
+    expect.assertions(1);
+    return rootTask.done.catch((e) => {
```

Comment:
I find `expect.assertions(N)` to be hard to maintain as apps grow. They also result in poor failure messages.

Here's a more robust way to do it:

```js
return rootTask.done
  .then(() => {
    throw new Error('unexpected success');
  })
  .catch((sagaError) => {
    mockApi.verify();
    expect(sagaError).toBe(error);
  });
```

btw, we try to avoid single character variable names in the code base.